app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -162,10 +223,119 @@ class TalkTopicActivity : BaseActivity() {
         }
     }
 
+    inner internal class ReplyTextWatcher: TextWatcher {
+        override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+        }
+
+        override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            reply_subject_layout.error = null
+            reply_text_layout.error = null
+        }
+
+        override fun afterTextChanged(p0: Editable?) {
+        }
+    }
+
+    private fun onSaveClicked() {
+        val subject = reply_subject_text.text.toString().trim()
+        var body = reply_edit_text.text.toString().trim()
+
+        if (isNewTopic() && subject.isEmpty()) {
+            reply_subject_layout.error = getString(R.string.talk_subject_empty)
+            reply_subject_layout.requestFocus()
+            return
+        } else if (body.isEmpty()) {
+            reply_text_layout.error = getString(R.string.talk_message_empty)
+            reply_text_layout.requestFocus()
+            return
+        }
+
+        // if the message is not signed, then sign it explicitly
+        if (!body.endsWith("~~~~")) {
+            body += " ~~~~"
+        }
+        if (!isNewTopic()) {
+            // add two explicit newlines at the beginning, to delineate this message as a new paragraph.
+            body = "\n\n" + body
+        }
```

Comment:
If there's no replies in a new topic, the first reply will stick with the topic.

Reply:
should be fixed.